doc/designs/external-idp/idp-api.md

```diff
@@ -0,0 +1,243 @@
+# IPA and an external identity provider integration - idp objects
+
+[TOC]
+
+IPA needs to store and manage IdP references. As a new IPA object, IdP reference needs:
+* creation of an LDAP object class and LDAP attribute types used to store IdP
+information
+* creation of a new IPA API managing the IdP references
+* definition of access control lists protecting access to the IdP references
+* upgrade and backward compatibility
+
+## LDAP Schema
+
+### New attribute types
+
+The external IdP reference object needs to contain the following information:
+* name (string)
+* authorization endpoint (URI mapped as a case-sensitive string),
+* device authorization endpoint (URI mapped as a case-sensitive string), for instance https://oauth2.googleapis.com/device/code
```

Comment:
really minor one: what about taking as an example a generic idp instead of google's one, this way we could add examples for:

auth endpoint, device auth endpoint, token endpoint, JWKS, userinfo endpoint

for instance https://${idpdomain}/oauth2/authorize 
for instance https://${idpdomain}/oauth2/code
for instance https://${idpdomain}/oauth2/token
...